src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,172 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * The TaskList class is a list of {@link Task}s.
+ * It supports the basic operations of manipulating tasks and answering queries.
+ */
+public class TaskList {
+    private List<Task> list;
```

Comment:
Should plural form be use for the name here as it is representing a collection of objects?